Source/SwiftLintCore/Extensions/Configuration+LintableFiles.swift

```diff
@@ -38,30 +38,41 @@ extension Configuration {
         excludeBy: ExcludeBy,
         fileManager: LintableFileManager = FileManager.default
     ) -> [String] {
+        func excludePaths(from includedPaths: [String]) -> [String] {
+            switch excludeBy {
+            case .prefix:
+                return filterExcludedPathsByPrefix(in: includedPaths)
+            case .paths(let excludedPaths):
+                return filterExcludedPaths(excludedPaths, in: includedPaths)
+            }
+        }
+
+        if path.isEmpty {
+            var pathsForPath = includedPaths
+            if pathsForPath.isEmpty {
+                pathsForPath = fileManager.filesToLint(inPath: path, rootDirectory: nil)
+            }
+            let includedPaths = pathsForPath
+                .flatMap(Glob.resolveGlob)
+                .parallelFlatMap { fileManager.filesToLint(inPath: $0, rootDirectory: rootDirectory) }
+            return excludePaths(from: includedPaths)
+        }
```

Comment:
This looks suspicious to me. `fileManager.filesToLint` (a very expensive operation) is called twice if there are no `includePaths`. This would be my idea of what this should look like:
```suggestion
        if path.isEmpty {
            if includedPaths.isEmpty {
                return excludePaths(from: fileManager.filesToLint(inPath: "", rootDirectory: rootDirectory))
            }
            let includedPaths = includedPaths
                .flatMap(Glob.resolveGlob)
                .parallelFlatMap { fileManager.filesToLint(inPath: $0, rootDirectory: rootDirectory) }
            return excludePaths(from: includedPaths)
        }
```

Reply:
Ah yes - that definitely does look better